### PR TITLE
dev-java/icu4j: bump to 70.1

### PR DESCRIPTION
**Pull request by vaukai**
Closes: https://bugs.gentoo.org/787803

 @fordfrog  Not yet ready.  

- It compiles but the resulting `jar` file still has differences compared with the upstream `jar` file.
- No tests yet (java-ebuilder failed to add them)
- The older ebuilds each provide 3 jar files. Clarification needed.

### CODE REVIEW

**Comment by vaukai** [comment]
Some classes removed and resources added, so the remaining differences compared with the [upstream jar file](https://github.com/unicode-org/icu/releases/download/release-69-1/icu4j-69_1.jar) might be ignored.

**Comment by vaukai** [comment]
TODOS

- [This stuff](https://github.com/gentoo/gentoo/pull/20661/commits/43c4471d0268b9f59c6ba1c08b6d430838dd877d?file-filters%5B%5D=.ebuild#diff-fdea08511c341ae2554c0a31ccaad78e4138ca60a22f0e9107f0b4241dea4662R50-R61) should not be deleted here in `src_unpack` but be used for creating the two missing jar files after [java-pkg-simple_src_compile](https://github.com/gentoo/gentoo/pull/20661/commits/43c4471d0268b9f59c6ba1c08b6d430838dd877d?file-filters%5B%5D=.ebuild#diff-fdea08511c341ae2554c0a31ccaad78e4138ca60a22f0e9107f0b4241dea4662R65) and then be `zip --delete`d from [icu4j.jar](https://github.com/gentoo/gentoo/pull/20661/commits/43c4471d0268b9f59c6ba1c08b6d430838dd877d?file-filters%5B%5D=.ebuild#diff-fdea08511c341ae2554c0a31ccaad78e4138ca60a22f0e9107f0b4241dea4662R69)
- There seem to be tests in `dev-java/icu4j`. Make them run. 
- Run `dev-util/pkgdiff`again for the new jar files and also for `dev-java/htmlparser` and `dev-java/jython`

**Comment by fordfrog** [comment]
@vaukai `dev-java/junitparams` merged, thanks!

**Comment by vaukai** [comment]
Back to the old design making much less trouble. No need for duplicate compilation of `icu4j.jar`.
@fordfrog 
Remaining problem is tests: [build.log.gz](https://github.com/gentoo/gentoo/files/6451500/build.log.gz) and here I again need help.

```
 * Compiling ...
warning: [options] bootstrap class path not set in conjunction with -source 8
tests/core/src/com/ibm/icu/dev/test/util/LocaleMatcherTest.java:27: error: package com.ibm.icu.dev.tool.locale does not exist
import com.ibm.icu.dev.tool.locale.LikelySubtagsBuilder;
                                  ^
tests/core/src/com/ibm/icu/dev/test/util/LocaleDistanceTest.java:15: error: package com.ibm.icu.dev.tool.locale does not exist
import com.ibm.icu.dev.tool.locale.LocaleDistanceBuilder;
                                  ^
tests/core/src/com/ibm/icu/dev/test/util/LocaleMatcherTest.java:880: error: cannot find symbol
        XLikelySubtags.Data built = LikelySubtagsBuilder.build();
                                    ^
  symbol:   variable LikelySubtagsBuilder
  location: class LocaleMatcherTest
tests/core/src/com/ibm/icu/dev/test/util/LocaleDistanceTest.java:53: error: cannot find symbol
        LocaleDistance.Data built = LocaleDistanceBuilder.build();
                                    ^
  symbol:   variable LocaleDistanceBuilder
  location: class LocaleDistanceTest
```

**Comment by vaukai** [comment]
Test errors reduced to [build.log.gz](https://github.com/gentoo/gentoo/files/6453365/build.log.gz)
```
 package com.sun.tools.doclets.formats.html.markup does not exist
 package com.sun.tools.doclets.internal.toolkit does not exist
 package com.sun.tools.doclets.internal.toolkit.taglets does not exist
```

File: dev-java/icu4j/icu4j-69.1.ebuild [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+
+JAVA_TEST_SRC_DIR=(
+	"tests"
+	"../tools/misc/src/com/ibm/icu/dev/tool/locale"
+)
+
+JAVA_TEST_RESOURCE_DIRS=(
+	"tests"
+	"tests/core/src/com/ibm/icu/dev/data"
+)
```
Comment (fordfrog) [inline dev-java/icu4j/icu4j-69.1.ebuild:51]: for the purpose of the ebuild, you should split test resources and test java classes as is done for maven projects. `java-pkg-simple` doesn't work correctly when sources and resources are mixed together.

**Comment by fordfrog** [comment]
@vaukai there are some comments in`icu4j` that seem obsolete and one todo? is that ready for review and merge?

**Comment by vaukai** [comment]
> @vaukai there are some comments in`icu4j` that seem obsolete and one todo? is that ready for review and merge?

@fordfrog Not ready, but needs help of an expert. Still test failures I cannot solve.  Once you run it as is it would fail with `"Tests run 5504, Failures: 39"`

**Comment by fordfrog** [comment]
@vaukai ok, will have a look at it :-)

**Comment by thesamesam** [comment]
ping, could we merge even if imperfect given Jython is uninstallable?

**Comment by fordfrog** [comment]
latest version is now 70.1

**Comment by vaukai** [comment]
> latest version is now 70.1

Just reworking the ebuild, will move to `70.1` also.  That's why in `draft` mode

**Comment by vaukai** [comment]
### Todo:

1. Solve remaining [29 test failures](https://gist.github.com/vaukai/56b4582cb0a59f6e9e55dd41b11ae0a2#file-icu4j-70-1-build-log-L752) presently test_excluded.
3. Lift `dev-java/jython-2.7.0-r3` to the new slot :70
4. Is `Main-Class` (`com.ibm.icu.util.VersionInfo`) needed or not (did not exist in older ebuilds)?

**Comment by vaukai** [comment]
### Todo:

1. Solve remaining [9 test failures](https://gist.github.com/vaukai/5181fb68de12e82a871532846bd34f79#file-icu4j-70-1-build-log-L752) presently test_excluded.
2. Is `Main-Class` (`com.ibm.icu.util.VersionInfo`) needed or not (did not exist in older ebuilds)?

**Comment by fordfrog** [comment]
@vaukai thanks! that was a tough one :-)

**Pull request closed without merging**